lib/rules/heuristic/shortname.js

```diff
@@ -15,34 +15,34 @@ exports.check = function (sr, done) {
     if (subType === 'member') topLevel = 'Submission';
     else if (subType === 'team') topLevel = 'TeamSubmission';
 
-    const extractLatestURI = function (dl) {
-        let latestAnchor;
-        if (dl) {
-            dl.querySelectorAll('dt').forEach((dt) => {
-                const txt = sr
-                    .norm(dt.textContent)
-                    .replace(':', '')
-                    .toLowerCase()
-                    .replace('published ', '');
-                const dd = util.next(dt, 'dd');
-                if (txt.lastIndexOf('latest version', 0) === 0)
-                    latestAnchor = dd && dd.querySelector('a');
-            });
-            return latestAnchor;
-        }
-    };
-
-    const linkLate = extractLatestURI(dl);
+    const dts = sr.extractHeaders(dl);
+    const linkLate = dts.Latest ? dts.Latest.dd.querySelector('a') : null;
     if (linkLate && linkLate.getAttribute('href') !== '') {
+        const linkLateHref = linkLate.getAttribute('href');
         const lateRex = `^https?:\\/\\/www\\.w3\\.org\\/${topLevel}\\/(.+?)\\/?$`;
-        const matches = (linkLate.getAttribute('href') || '')
-            .trim()
-            .match(new RegExp(lateRex));
+        const matches = linkLateHref.trim().match(new RegExp(lateRex));
+        let shortname = '';
         if (matches) {
-            const sn = matches[1];
-            sr.metadata('shortname', sn);
+            [, shortname] = matches;
+            sr.metadata('shortname', shortname);
         }
-    } else sr.error(self, 'latest-link');
 
-    return done();
+        // make sure FPWD doesn't use a shortname that exists.
+        const isFP = sr.config && !sr.config.previousVersion;
+        if (shortname && isFP) {
+            req.get(linkLateHref, {}, (error, response) => {
```

Comment:
If you only care about the http code, `head` should be better

```suggestion
            req.head(linkLateHref, {}, (error, response) => {
```